rust/src/milla/api.rs

```diff
@@ -272,25 +283,35 @@ pub(crate) fn internal_set_tile(
     if let Some(value) = thermal_energy {
         tile.thermal_energy = value;
     }
+    if let Some(value) = hotspot_temperature {
+        tile.hotspot_temperature = value;
+    }
+    if let Some(value) = hotspot_volume {
+        tile.hotspot_volume = value;
+    }
 
     Ok(())
 }
 
 /// BYOND API for fetching the atmos details of a tile.
 #[byondapi::bind]
-fn milla_get_tile(turf: ByondValue, list: ByondValue) {
+fn milla_get_tile(turf: ByondValue, list: ByondValue) -> eyre::Result<ByondValue> {
     logging::setup_panic_handler();
     let (x, y, z) = byond_xyz(&turf)?.coordinates();
     let tile = internal_get_tile(x as i32 - 1, y as i32 - 1, z as i32 - 1)?;
     let vec: Vec<ByondValue> = (&tile).into();
     list.write_list(vec.as_slice())?;
-    Ok(Default::default())
+    Ok(ByondValue::null())
 }
 
 /// Rust version of fetching the atmos details of a tile.
 pub(crate) fn internal_get_tile(x: i32, y: i32, z: i32) -> Result<Tile> {
     let buffers = BUFFERS.get().ok_or(eyre!("BUFFERS not initialized."))?;
-    let active = buffers.get_active().read().unwrap();
+    let maybe_active = buffers.get_active().read();
+    if maybe_active.is_err() {
+        return Err(eyre!("MILLA buffers have been poisoned, MILLA cannot run."));
```

Comment:
Is this unrecoverable? If so, we may want to pass it back as a value that BYOND can act on, possibly marking SSair as offline and recommending a shuttle call to admins.

Reply:
Yeah. Adjusted what happens here to report breathable air, and updated the error callback to kill SSair and report the error to admins.